hack/test.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+junitFilenamePrefix() {
+  if [[ -z "${JUNIT_REPORT_DIR}" ]]; then
+    echo ""
+    return
+  fi
+  mkdir -p "${JUNIT_REPORT_DIR}"
+  DATE=$( date +%s | base64 | head -c 15 )
+  echo "${JUNIT_REPORT_DIR}/junit_$DATE"
+}
+
+function produce_junit_xmlreport {
+  local -r junit_filename_prefix=$1
+  if [[ -z "${junit_filename_prefix}" ]]; then
+    return
+  fi
+
+  local junit_xml_filename
+  junit_xml_filename="${junit_filename_prefix}.xml"
+
+  if ! command -v gotestsum >/dev/null 2>&1; then
+    echo "gotestsum not found; installing now"
+    go install gotest.tools/gotestsum@latest
+  fi
+  ./bin/gotestsum --junitfile "${junit_xml_filename}" --raw-command cat "${junit_filename_prefix}"*.stdout
```

Comment:
what's the guaranty that this binary will be in `./bin` dir? because `go install gotest.tools/gotestsum@latest` will install in the GOPATH and that can be anywhere!

Reply:
Right the binary may not always be in `./bin` dir!

While running this from a prow pod, it had `GOBIN` env variable set to current work_dir and hence the GOPATH was overwritten. (It did not install in $GOPATH)
I assumed this will only run from prow pods and used ./bin/gotestsum. 👎 

I shall add `$GOBIN` and `$GOPATH/bin` to the PATH variable and directly run the binary. @mkumatag hope that will be fine.

Reply:
If this is part of makefile then anyone can run from the local laptop to test it. Yes, adding the bin path will be helpful.

Reply:
Done.